src/DatePicker/DatePicker.tsx

```diff
@@ -6,50 +6,35 @@ import { props, t, ReactChild } from "../utils";
 import * as moment from "moment";
 
 export namespace DatePicker {
-  export type Value = string | Date | moment.Moment;
-  export type OnChangeProps<T extends Value> =
-    | {
-        /** MomentJS format used to format date before returing through "onChange" */
-        returnFormat?: never;
-        /** called when value changes */
-        onChange?: (date?: T) => void;
-      }
-    | {
-        /** MomentJS format used to format date before returing through "onChange" */
-        returnFormat: string;
-        /** called when value changes */
-        onChange?: (date?: string) => void;
-      };
-
-  export type Props<T extends Value> = {
+  export type Props = {
     /** an optional id to pass to top level element of the component */
     id?: string;
     /** current date */
-    value?: T;
+    value?: Date;
     /** default date */
-    defaultValue?: T;
+    defaultValue?: Date;
+    /** called when value changes */
+    onChange?: (date?: Date) => void;
     /** called when datepicker is closed */
     onHide?: () => void;
     /** MomentJS format used to display current date */
     displayFormat?: string;
-    /** called when value is cleared */
-    onClear?: () => void;
     /** minimum date selectable by the user */
-    minDate?: Value;
+    minDate?: Date | string;
```

Comment:
is there a reason to also keep `string` in input? Can't we just stick to `Date`

Reply:
no particular reason. I left it just for convenience: there's no complex logic around these inputs and you have to convert it to a `Moment` date anyway, so I decided to keep everything that can be passed to `moment()`.
btw I'm ok to remove them if you prefer.

Reply:
I prefer removing it. Having a stricter interface will make our projects more consistent and it allows changing the internals more freely.